### DISCUSSION TITLE
How to update existing WPF app to WinUI 3 fluent components?

### DISCUSSION THREAD

**Discussion by steam3d** [microsoft/WindowsAppSDK#1005] [Q&A] [answered]
I can't find any information how to use this SDK with existing app. (Only how to [create a new app](https://docs.microsoft.com/ru-ru/windows/apps/winui/winui3/create-your-first-winui3-app?tabs=desktop-csharp))

**Comment by MEK3DK** [comment, marked as answer]
You will need to use Xaml Islands in WPF to host WinUI 3.

Xaml Islands are however not ready yet, MS aims at a experimental release in Q4 of this year and maybe something stable in Q1 2022, but it is a complex feature.

Without Xaml Islands you are pretty much stuck on WPF, Xaml Islands is a must have feature.

**Reply by maxkatz6** [reply]
> Without Xaml Islands you are pretty much stuck on WPF

Reimplementing specific windows in WinUI is possible though.

**Reply by steam3d** [reply]
So if i have a simple WPF app the best way just rewrite the app on Project Reunion?

**Reply by maxkatz6** [reply]
Imo, you will need to do it anyway eventually if you want to maintain your application in long term.
People are saying that UWP is dead, but it's pretty much more alive than WPF. Though WPF has bigger community and controls libraries base.

**Reply by steam3d** [reply]
Thanks for the help

**Reply by MEK3DK** [reply]
> Imo, you will need to do it anyway eventually if you want to maintain your application in long term.

Completely agree, in time you app should turn into full WinUI 3, but if it is a big app then you have to do it step by step. 
Luckily the difference between WPF -> WinUI 3 is not as big as WinForm -> WPF. That required a lot of rewrite.

> People are saying that UWP is dead, but it's pretty much more alive than WPF. Though WPF has bigger community and controls libraries base.

When WinUI is done I have to agree with those people, I don't really see the use case for UWP. For desktop apps there was never really a use case for UWP because you were in a sandbox, thus WPF has been the only option for many years.
But yes sadly Microsoft did not maintain/improve WPF, it was a UI layer that had so much potential so many years ago. Imagine if MS had done for WPF 8-10 years what they are doing with WinUI today.
